### ISSUE TITLE
NPE in EntityModelLoader when entity model is invalid

### ISSUE THREAD

**Issue by s4heid** [SAP/neonbee#199]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Given an invalid model, the EntityModelLoader does not properly validate the model. Instead of propagating the error and failing the deployment of the model, a NullPointerException is thrown, which is not catched.

The problematic code is in [`EntityModelLoader.buildModelMap`](https://github.com/SAP/neonbee/blob/cd6ccd993ae1310937f0c85162b191325f825b1f/src/main/java/io/neonbee/entity/EntityModelLoader.java#L204-L206), because `serviceMetaData.getEdm().getEntityContainer()` might be null.

### Expected Behavior

The deployment of the invalid model should fail with an error message, describing the issue.

[Example code](https://github.com/SAP/neonbee/commit/fe7b486585ab995f3118433a1531db089bbd224e), which futurizes `buildModelMap` and handles the error:

```
20:00:16.888 [vert.x-eventloop-thread-10] ERROR io.neonbee.internal.deploy.PendingDeployment - Deployment of Models(models/data-model.csn$models/Esrc.Service.edmx) failed
io.vertx.core.impl.NoStackTraceThrowable: Cannot build model map. Entity Container is empty.
20:00:16.903 [vert.x-eventloop-thread-6] INFO io.neonbee.internal.verticle.ServerVerticle - HTTP server started on port 8080
20:00:16.904 [vert.x-eventloop-thread-10] ERROR io.neonbee.internal.deploy.PendingDeployment - Deployment of Deployables(module:2.0.0-SNAPSHOT) failed
io.vertx.core.impl.NoStackTraceThrowable: Cannot build model map. Entity Container is empty.
20:00:16.904 [vert.x-eventloop-thread-10] ERROR io.neonbee.internal.verticle.DeployerVerticle - Unexpected error occurred during deployment of module from JAR file: /Users/d061008/workspace/neonbee/working_dir/modules/dummy-2.0.0-SNAPSHOT-models.jar
io.vertx.core.impl.NoStackTraceThrowable: Cannot build model map. Entity Container is empty.
```

### Steps To Reproduce

Clone neonbee from the `main` branch, create a folder `working_dir/modules`, and create an invalid cds / edmx file. Start neonbee with `./gradlew run` and you should be able to reproduce the error.

### Environment

```markdown
- OS: Darwin / 21.6.0 / arm64
- Java: OpenJDK Runtime Environment SapMachine (build 11.0.14.1+1-LTS-sapmachine)
- NeonBee: 0.16.1
```

### Relevant log output

```shell
java.lang.NullPointerException: Cannot invoke "org.apache.olingo.commons.api.edm.EdmEntityContainer.getNamespace()" because the return value of "org.apache.olingo.commons.api.edm.Edm.getEntityContainer()" is null
        at io.neonbee.entity.EntityModelLoader.lambda$buildModelMap$16(EntityModelLoader.java:205)
        at java.base/java.util.stream.Collectors.lambda$uniqKeysMapAccumulator$1(Collectors.java:177)
        at java.base/java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
        at io.neonbee.entity.EntityModelLoader.buildModelMap(EntityModelLoader.java:204)
        at io.neonbee.entity.EntityModelLoader.lambda$loadModel$11(EntityModelLoader.java:184)
        at io.vertx.core.impl.future.FutureImpl$1.onSuccess(FutureImpl.java:91)
        at io.vertx.core.impl.future.FutureImpl$ListenerArray.onSuccess(FutureImpl.java:262)
        at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
        at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
        at io.vertx.core.impl.future.CompositeFutureImpl.trySucceed(CompositeFutureImpl.java:163)
        at io.vertx.core.impl.future.CompositeFutureImpl.lambda$all$0(CompositeFutureImpl.java:38)
        at io.vertx.core.impl.future.FutureImpl$3.onSuccess(FutureImpl.java:141)
        at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
        at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
        at io.vertx.core.impl.future.Composition$1.onSuccess(Composition.java:62)
        at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:60)
        at io.vertx.core.impl.future.FutureImpl.tryComplete(FutureImpl.java:211)
        at io.vertx.core.impl.future.PromiseImpl.tryComplete(PromiseImpl.java:23)
        at io.vertx.core.impl.future.PromiseImpl.onSuccess(PromiseImpl.java:49)
        at io.vertx.core.impl.future.FutureBase.lambda$emitSuccess$0(FutureBase.java:54)
        at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:829)
```

**Comment by kristian** [comment]
Confirmed, CDS files with no services configured:

```cds
namespace Test;

service Service {
    // empty
}

entity TestEntity {
	name : String
}
```

Do indeed generate EDMX files w/O an entity container:

```xml
<?xml version="1.0" encoding="utf-8"?>
<edmx:Edmx Version="4.0" xmlns:edmx="http://docs.oasis-open.org/odata/ns/edmx">
  <edmx:DataServices>
    <Schema Namespace="Test.Service" xmlns="http://docs.oasis-open.org/odata/ns/edm"/>
  </edmx:DataServices>
</edmx:Edmx>
```

The CDS compiler spits out a warning about that:

```
[WARNING] test.cds:3:9: Schema “Test.Service” is empty (in service:"Test.Service")
```

EDMX files w/o an `EntityContainer` element should get ignored in the `EntityModelLoader`.

**Comment by kristian** [comment]
Fixed by #201, feel free to review

**Comment by kristian** [comment]
Fixed with #201
